pkg/machinepool/machinepool.go

```diff
@@ -280,3 +291,691 @@ func getNodePoolsString(nodePools []*cmv1.NodePool) string {
+		if interactive.Enabled() {
+			autoscaling, err = interactive.GetBool(interactive.Input{
+				Question: "Enable autoscaling",
+				Help:     cmd.Flags().Lookup("enable-autoscaling").Usage,
+				Default:  autoscaling,
+				Required: false,
+			})
+			if err != nil {
+				err = fmt.Errorf("Expected a valid value for enable-autoscaling: %s", err)
+				return
+			}
+		}
+	}
+
+	if autoscaling {
+		// Prompt for min replicas if neither min or max is set or interactive mode
+		if !isMinReplicasSet && (interactive.Enabled() || !isMaxReplicasSet && !isAnyAdditionalParameterSet) {
+			minReplicas, err = interactive.GetInt(interactive.Input{
+				Question: "Min replicas",
+				Help:     cmd.Flags().Lookup("min-replicas").Usage,
+				Default:  existingAutoscaling.MinReplica(),
+				Required: replicasRequired,
+				Validators: []interactive.Validator{
+					mpHelpers.MinNodePoolReplicaValidator(true),
+				},
+			})
+			if err != nil {
+				err = fmt.Errorf("Expected a valid number of min replicas: %s", err)
+				return
+			}
+		}
+
+		// Prompt for max replicas if neither min or max is set or interactive mode
+		if !isMaxReplicasSet && (interactive.Enabled() || !isMinReplicasSet && !isAnyAdditionalParameterSet) {
+			maxReplicas, err = interactive.GetInt(interactive.Input{
+				Question: "Max replicas",
+				Help:     cmd.Flags().Lookup("max-replicas").Usage,
+				Default:  existingAutoscaling.MaxReplica(),
+				Required: replicasRequired,
+				Validators: []interactive.Validator{
+					mpHelpers.MaxNodePoolReplicaValidator(minReplicas),
+				},
+			})
+			if err != nil {
+				err = fmt.Errorf("Expected a valid number of max replicas: %s", err)
+				return
+			}
+		}
+	} else if interactive.Enabled() || !isReplicasSet {
+		if !isReplicasSet {
+			replicas = existingReplicas
+		}
+		if !interactive.Enabled() && isAnyAdditionalParameterSet {
+			// Not interactive mode and we have at least an additional parameter set, just keep the existing replicas
+			return
+		}
+		replicas, err = interactive.GetInt(interactive.Input{
+			Question: "Replicas",
+			Help:     cmd.Flags().Lookup("replicas").Usage,
+			Default:  replicas,
+			Required: true,
+			Validators: []interactive.Validator{
+				mpHelpers.MinNodePoolReplicaValidator(false),
+			},
+		})
+		if err != nil {
+			err = fmt.Errorf("Expected a valid number of replicas: %s", err)
+			return
+		}
+	}
+	return
+}
+
+func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int) *cmv1.NodePoolAutoscalingBuilder {
+	existingMinReplica := nodePool.Autoscaling().MinReplica()
+	existingMaxReplica := nodePool.Autoscaling().MaxReplica()
+
+	min := existingMinReplica
+	max := existingMaxReplica
+
+	if minReplicas != 0 {
+		min = minReplicas
+	}
+	if maxReplicas != 0 {
+		max = maxReplicas
+	}
+
+	if existingMinReplica != minReplicas || existingMaxReplica != maxReplicas {
+		if min >= 1 && max >= 1 {
+			return cmv1.NewNodePoolAutoscaling().MinReplica(min).MaxReplica(max)
+		}
+	}
+
+	return nil
+}
```

Comment:
I made a couple of changes, which was just grabbing the args via cobra flags, as this was the first refactor which used the args directly in these functions. Otherwise, we have just been grabbing machinepool/nodepool ID + cluster in the args, which is included in the runtime + parameters for each function

If I include all the args for `edit machinepool`, the parameters for the functions would be quite long.

I debated making a struct to contain this information, but this file is already 1000 lines long, so i am at a crossroads:

1. no struct, keep as is (using cmd.flag("FLAG_NAME").value to get the arguments)
2. separate nodepool and machinepool funcs into their own files (preferred) and use a struct (potentially a 3rd file for data types)

Reply:
lets not make this PR any bigger. Keep as is for now. If you want to follow up with your suggestion in a different MR that would be good